sound/soc/intel/boards/Kconfig

```diff
@@ -304,7 +304,7 @@ config SND_SOC_INTEL_BXT_DA7219_MAX98357A_MACH
 	tristate "Broxton with DA7219 and MAX98357A in I2S Mode"
 	depends on I2C && ACPI
 	depends on MFD_INTEL_LPSS || COMPILE_TEST
-	depends on SND_HDA_CODEC_HDMI
+	depends on SND_HDA_CODEC_HDMI && SND_SOC_SOF_HDA_AUDIO_CODEC
```

Comment:
@libinyang Actually this is a mistake in my earlier patch. This mach driver is not used with SOF, so it shouldn't have SND_HDA_CODEC_HDMI.

Reply:
@kv2019i Do you mean we should remove `SND_HDA_CODEC_HDMI` in the Kconfig? I checked and found soc-acpi-intel-bxt-match.c has already added `.sof_fw_filename = "sof-apl.ri"` and `.sof_tplg_filename = "sof-apl-da7219.tplg"` in `snd_soc_acpi_intel_bxt_machines`. Does this mean our SOF has already supported it? If we remove `SND_HDA_CODEC_HDMI`, I think we should also remove some code related to the common hdmi.

Reply:
@libinyang Good question. I saw Pierre actually removed SOF support for this BXT driver with commit "ASoC: Intel: boards: remove support for SOF for APL devices" (this is after my common-hdmi series). So I would not touch this machine driver in this patch.

@plbossart Can you recall what is the history? We probably should have removed SND_HDA_CODEC_HDMI depends when we removed SOF support.

Reply:
@kv2019i Got it. I will remove my fix for this machine driver. Thanks.

Reply:
This machine driver is used for APL Chromebooks. We cannot change the firmware key to use the SOF public one except on preproduction hardware. Seppo and I are the only ones to have such devices. 

Now that the firmware is signed with the production key, we could add SOF support here. I am not sure if we want to, or this would be something to discuss separately. Let's leave it as is for now.

Reply:
@plbossart Thanks for the information. I will not touch SND_SOC_INTEL_BXT_DA7219_MAX98357A_MACH and SND_SOC_INTEL_BXT_RT298_MACH in this patch.